calico_kubernetes/calico_kubernetes_cni.py

```diff
@@ -307,6 +290,297 @@ def _get_container_pid(container_id):
+                # Replace arg 'label' with arg 'tag'
+                label_ind = args.index('label')
+                args[label_ind] = 'tag'
+
+                # Split given label 'key=value' into components 'key', 'value'
+                label = args[label_ind + 1]
+                key, value = label.split('=')
+
+                # Compose Calico tag out of key, value components
+                tag = _label_to_tag(key, value)
+                args[label_ind + 1] = tag
+
+            # Remove empty strings and add to rule list
+            args = filter(None, args)
+            inbound_rules.append(args)
+
+    return inbound_rules, outbound_rules
+
+
+def _label_to_tag(ns, label_key, label_value):
+    """
+    Labels are key-value pairs, tags are single strings. This function handles that translation
+    1) Concatenate key and value with '='
+    2) Prepend a pod's namespace followed by '/' if available
+    3) Escape the generated string so it is Calico compatible
+
+    :param label_key: key to label
+    :param label_value: value to given key for a label
+    :param namespace: Namespace string, input None if not available
+    :param types: (self, string, string, string)
+    :return single string tag
+    :rtype string
+    """
+    tag = '%s=%s' % (label_key, label_value)
+    tag = '%s/%s' % (ns, tag)
+    tag = _escape_chars(tag)
+    return tag
+
+
+def _get_namespace_tag(namespace):
+    """
+    Pull metadata for namespace and return it and a generated NS tag
+    """
+    ns_tag = _escape_chars('%s=%s' % ('namespace', ns))
+    return ns_tag
+
+
+def _escape_chars(self, unescaped_string):
+    """
+    Calico can only handle 3 special chars, '_.-'
+    This function uses regex sub to replace SCs with '_'
+    """
+    # Character to replace symbols
+    swap_char = '_'
+
+    # If swap_char is in string, double it.
+    unescaped_string = re.sub(swap_char, "%s%s" % (swap_char, swap_char), unescaped_string)
+
+    # Substitute all invalid chars.
+    return re.sub('[^a-zA-Z0-9\.\_\-]', swap_char, unescaped_string)
+
+
+def _get_metadata(namespace, pod_name, key):
+    """Return the Metadata[key] of the pod to which the container belongs to"""
+    pod_info = _get_pod_config(namespace, pod_name)
+    return pod_info["metadata"][key]
+
+
+def _get_pod_config(namespace, pod_name):
+     """
+     Get the list of pods from the Kube API server.
+     """
+     pods = _get_api_path('pods')
+     _log.debug('Got pods %s' % pods)
+
+     for pod in pods:
+         _log.debug('Processing pod %s', pod)
+         if pod['metadata']['namespace'].replace('/', '_') == namespace and \
+                         pod['metadata']['name'].replace('/', '_') == pod_name:
+             this_pod = pod
+             break
+         else:
+             raise KeyError('Pod %s not found', pod_name)
+     _log.debug('Got pod data %s', this_pod)
+     return this_pod
+
+
+def _get_api_path(path):
+    """Get a resource from the API specified API path.
```

Comment:
Pep 8 suggests a standard here: https://www.python.org/dev/peps/pep-0257/.

Option 1 above is ok (we do it in plenty of places), but 2 above is more nuanced. If you put text on the first line, it specifically should be the title summary of the function.

E.g. 
def foobar():
    """Foo the bar
    What's a foo? What's a bar? etc.
    """

I see fairly often folk doing
    """start my 
    line and continue it
    """
Which isn't going to render well in e.g. the Python help() command.